src/esp/sensor/CameraSensor.h

```diff
@@ -23,14 +23,9 @@ class CameraSensor : public VisualSensor {
 
   void setProjectionParameters(const SensorSpec::ptr& spec);
 
-  // set the projection matrix to the given render camera
-  virtual CameraSensor& setProjectionMatrix(
-      gfx::RenderCamera& targetCamera) override;
-  // set the transformation matrix to the given render camera
-  virtual CameraSensor& setTransformationMatrix(
-      gfx::RenderCamera& targetCamera) override;
-  // set the view port to the given render camera
-  virtual CameraSensor& setViewport(gfx::RenderCamera& targetCamera) override;
+  virtual gfx::RenderCamera* getRenderCamera() override;
+
+  auto computeTransformationMatrix();
```

Comment:
I am wondering why you still need this function??

To answer this question, let us first look at another one:
Why do we need a function to set the transformation matrix before? 
Because the camera (the `defaultRenderCamera`) was responsible for the rendering requests from all the sensors. For example, sensor A asked it to render the scene (so you have to set the render camera based on the sensor A's spec), then, sensor B asked it to render the scene, so you have to refresh the render camera settings with the spec from sensor B.
![old-camera-design](https://user-images.githubusercontent.com/931093/105565064-d4a25880-5cd9-11eb-86da-51ce9f13a999.png)
[Figure illustrates the OLD design]

Now, after this refactoring, this is not the case. Each sensor has its own embedded camera! It moves as the sensor moves, which means the transformation matrix of the render camera is always up to date. You do not have to set it every time in the `getObeservation()` (namely, you do not need: `renderCamera_->node().setTransformation(this->computeTransformationMatrix());`)

And I do not think you need this `computeTransformationMatrix` at all! It can be fully deprecated.